design-proposals/host-ip-pool.md

```diff
@@ -0,0 +1,75 @@
+# Host IP pooling with NNCPs.
+
+## Summary
+
+Provided a mechanism to configure automatically static IP configuration for secondary interfaces on existing and future Nodes
+
+## Motivation
+
+Using DHCP as the IP assignment mechanism has some drawbacks:
+- Introduce latencies on node restart.
+- Has problematic behavious when the DHCP server is miss configured.
+- It depends on network setup that can be complex.
+
+To overcome this static configured interfaces taking the address from a
+dynamic host IP pool is a solution. Since the interfaces are static there is no
+reconfiguration after some time or at node restart.
+
+This functionality was requested on U/S via https://github.com/nmstate/kubernetes-nmstate/issues/725
+
+Spike using whereabouts https://github.com/nmstate/kubernetes-nmstate/pull/1081
+
+### User Stories
+
+- As a cluster administrator, I want to configure static IPs on secondary NICs on a pool of nodes without creating one per each node.
+- As a cluster administrator, I want scale up the cluster with new nodes getting IPs from the policy.
+- As a cluster administrator, I want each of these static IPs to be unique, and taken from a defined range, and freed when unused.
+
+### Goals
+
+- Allow users to use NodeNetworkConfigurationPolicy to configure static cluster-unique IP addresses without hard-coding them.
+- If at all possible, we should pick a solution that could be utilized even in the IPI nmstate API.
+
+### Non-Goals
+
+- This design does not cover IP address configuration of management NIC on day-0/1
+
+## Proposal
+## Implementation History
+
+- Day 2 ippool with kubernetes-nmstate is implemented.
+- The ippool library at kubernetes-nmstate is extracted or make it public.
+- Baremetal IPI is implemented adding `networkPolicy` field at install config.
```

Comment:
I can't make any sense of this proposal with just this info.

Reply:
Re-worked it